### PR TITLE
Add RegExp support to ChangeTagValue recipe

### PR DESCRIPTION
**Pull request by anthochristen**
## What's changed?
Allows ChangeTagValue to take in regex in `oldValue` parameter.

## What's your motivation?
- Fixes https://github.com/openrewrite/rewrite/issues/4731

## Anything in particular you'd like reviewers to focus on?

Please leave your thoughts here or on the issue https://github.com/openrewrite/rewrite/issues/4731. 

## Anyone you would like to review specifically?
@timtebeek? - Because I see you actively replying ;) 
@gadams00 - Because, history.

## Have you considered any alternatives or workarounds?

Not really. I found this to be more appealing.

### Checklist
- [x] I've added unit tests to cover both positive and negative cases
- [x] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [x] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

**Comment by anthochristen** [comment]
@timtebeek / @gadams00 Can you kindly help with this change? Thanks!

File: rewrite-xml/src/test/java/org/openrewrite/xml/ChangeTagValueTest.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            <dependency>
+              <group>com.company.project</group>
+              <group>artifact</group>
+              <version>1.3.4</version>
+            </dependency>
+            """, spec -> spec.path("pom.xml")
+          )
+        );
+    }
+}
```
[the remark this answers is not in the document]
Comment (anthochristen) [reply rewrite-xml/src/test/java/org/openrewrite/xml/ChangeTagValueTest.java:122]: Unsure what the change here is. Seems moot, closing it.

**Comment by anthochristen** [comment]
@timtebeek / @gadams00 Any thoughts on this PR?

**Comment by timtebeek** [comment]
Thanks for the work here so far @anthochristen ! Didn't get to a review earlier beyond the automated suggestions, thanks for applying those already! From https://github.com/openrewrite/rewrite/issues/4731 I understand you want to use this from a declarative recipe to update a tag at a specific XPath ? Seems fair enough.

I see you've taken care to add backwards compatible constructors; that's appreciated but perhaps not necessary. Typically we roll forward by running this recipe against the OpenRewrite org
- https://app.moderne.io/recipes/org.openrewrite.java.AddNullMethodArgument
- https://app.moderne.io/devcenter/OpenRewrite

The method pattern in this case would be `org.openrewrite.xml.ChangeTagValue <constructor>(..)` once this PR is merged.
Feel free to remove those backwards compatible constructors and then we can merge this in.

**Comment by anthochristen** [comment]
> Thanks for the work here so far @anthochristen ! Didn't get to a review earlier beyond the automated suggestions, thanks for applying those already! From #4731 I understand you want to use this from a declarative recipe to update a tag at a specific XPath ? Seems fair enough.
> 
> I see you've taken care to add backwards compatible constructors; that's appreciated but perhaps not necessary. Typically we roll forward by running this recipe against the OpenRewrite org

Thanks @timtebeek for reviewing the change. Yes, thats the purpose of issuing this PR. To do find and replace for a specific X-Path, using the existing declarative `ChangeTagValue` recipe.

About removing the old constructors, was un-sure if that'd amount to semver issues. So refactored my regex changes as a separate visitor without affecting the old one. 

Hope that works? If not tell me know I can revert this and remove the constructors, as suggested, & update the usages.

**Review by timtebeek — APPROVED** [review]
Thanks for the work done here @anthochristen ! Took a bit longer to review than I had liked, but really appreciate the thorough tests you added where those were missing before. They made refactoring easy for me, which I did to limit the publicly visible API after this change. The visitor you added was only ever used when regex was true, so then I figured it easier to make that the dedicated case, as opposed to have two ways to do non-regex replacements.

**Pull request merged**